src/filters/filters.js

```diff
@@ -5,12 +5,36 @@ import xGovFilters from '@x-govuk/govuk-prototype-filters'
 
 const { govukMarkdown } = xGovFilters
 
-const addFilters = (nunjucksEnv) => {
+/**
+ *
+ * @param {*} dataSubjects
+ * @returns {Map<string,string>}
+ */
+function createDatasetMapping (dataSubjects) {
+  const mapping = new Map()
+  for (const data of Object.values(dataSubjects)) {
+    for (const dataset of data.dataSets) {
+      mapping.set(dataset.value, dataset.text)
+    }
+  }
+  return mapping
+}
+
+const addFilters = (nunjucksEnv, { dataSubjects }) => {
   nunjucksEnv.addFilter('govukMarkdown', govukMarkdown)
   nunjucksEnv.addFilter('getkeys', getkeys)
   nunjucksEnv.addFilter('getContext', getContext)
   nunjucksEnv.addFilter('toErrorList', toErrorList)
   nunjucksEnv.addFilter('validationMessageLookup', validationMessageLookup)
+
+  const datasetNameMapping = createDatasetMapping(dataSubjects)
+  nunjucksEnv.addFilter('datasetSlugToReadableName', function (slug) {
+    const name = datasetNameMapping.get(slug)
+    if (!name) {
+      throw new Error(`Can't find a name for ${slug}`)
+    }
+    return name
+  })
```

Comment:
should we move this into its own file similar to the other filters that are added?

Reply:
I didn't do it, because it's only one filter, the others are debugging related, so there's reason to group them